mender-dist-packages-building/install-armv6-cross-toolchain

```diff
@@ -0,0 +1,170 @@
+    local -r pkg=$2
+    deb_package_url=$(grep Filename Packages | grep /${pkg}_ | grep armhf | tail -n1 | sed 's/Filename: //');
+    filename=$(basename $deb_package_url);
+    curl --location http://raspbian.raspberrypi.org/raspbian/${deb_package_url} -o $filename 2>/dev/null || { echo "Did not find the package: ${pkg} in the raspbian repos.."; exit 1; }
+};
+
+function install_cross_compiler_toolchain() {
+    # Install the armv6 supported cross toolchain from Kotlin
+    [[ $# -ne 3 ]] && { echo "Three arguments required. got $#"; exit 1; }
+    local -r install_prefix="$1"
+    local -r download_url="$2"
+    local -r toolchain_version="$3"
+    echo "Installing the cross compiler toolchain to ${install_prefix}<cross-compiler>"
+    (
+        cd "$install_prefix"
+        wget -nc -q ${download_url}${toolchain_version}.tar.bz2
+        # wget -nc -q https://toolchains.bootlin.com/downloads/releases/toolchains/armv6-eabihf/tarballs/armv6-eabihf--glibc--stable-2020.08-1.tar.bz2
+        tar -xjf ${toolchain_version}.tar.bz2
+        # tar -xjf armv6-eabihf--glibc--stable-2020.08-1.tar.bz2
+        rm ${toolchain_version}.tar.bz2;
+        # rm armv6-eabihf--glibc--stable-2020.08-1.tar.bz2;
+    )
+}
+
+# Get depdendencies from upstream, manually donwloading deb packages, and fake pkg-config.
+# NOTE: pkg-config is used from go build to obtain cflags and libs required to build C code; however
+# for armhf we will explicitly pass these flags for go to use our custom sysroot instead of the system
+# one. Alternatively pkg-config supports setting a prefix, but that would require patching the source code.
+function setup_pkg_config() {
+    local -r URL="$1"
+    ln -s /bin/true /usr/bin/pkg-config
+    curl -f "${URL}" -o Packages.gz
+    gunzip Packages.gz
+}
+
+function install_packages_to_toolchain_root() {
+    local -r toolchain_root="$1"; shift
+    local -r required_packages_="$@"
+    for pkg in ${required_packages_} ; do
+        download_pkg ${toolchain_root} ${pkg}
+        echo "Extracting ${pkg} to an archive (tar.(xz|gz))...";
+        ar -x ${pkg}_*_armhf.deb data.tar.xz;
+        for dir in ./lib ./usr/include ./usr/lib; do
+            tar --list --file=data.tar.xz ${dir} >& /dev/null && tar --extract  --directory ${toolchain_root}/arm-buildroot-linux-gnueabihf/sysroot/ --file=data.tar.xz --transform='s,/arm-linux-gnueabihf,,'  ${dir} || true
+        done
+        echo "Finished ${pkg}"
+    done
+}
+
+function install_libffi() {
+    if cat "$2" | grep -E "^Package: libffi7$"; then
+        install_packages_to_toolchain_root "$1" libffi7;
+    else
+        install_packages_to_toolchain_root "$1" libffi6;
+    fi;
+}
+
+
+#
+# main
+#
+
+install_cross_compiler_toolchain "${TOOLCHAIN_PREFIX}" "${BOOTLIN_URL}" "${ARMV6_TOOLCHAIN}"
+
+setup_pkg_config "${DEBIAN_URL}"
+
+install_packages_to_toolchain_root "${ARMV6_TOOLCHAIN_ROOT}" ${required_packages[@]}
+
+# Depends on which version is required (6 || 7)
```

Comment:
This comment makes more sense _inside_ the function.

Reply:
I just removed it